### PR TITLE
FI-465: Reorganize tests

### PR DESCRIPTION
**Pull request by Jammjammjamm**
- Move modules from `lib/app/modules` to `lib/modules`
- Move sequence tests from `test/sequence` to `lib/module/some_module/test`
- Move tests from `test/model` to `test/unit`
- Update `Rakefile` so `rake` runs tests in test directory and module directories
- Update `test_helper` so that fixture load path is determined dynamically

The code climate issues aren't really new, they just appear as new because the files were moved.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-465
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/argonaut/test/argonaut_diagnostic_report_sequence_test.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-require_relative '../../test_helper'
+require_relative '../../../../test/test_helper'
```
Comment (radamson) [inline lib/modules/argonaut/test/argonaut_diagnostic_report_sequence_test.rb]: Is there a better place for `test_helper`?

Comment (Jammjammjamm) [reply lib/modules/argonaut/test/argonaut_diagnostic_report_sequence_test.rb]: This is ugly, but the tests in `test` and in `lib/modules/some_module/test` both need access to it. I tried to have rake load it, but I couldn't get that to work, perhaps because rake runs in a different process or something. So I'm just leaving it where it is for now.
